src/lib/context.js

```diff
@@ -8,6 +8,21 @@ export class Context {
   constructor(env, resIds) {
     this._env = env;
     this._resIds = resIds;
+    this._numberFormatters = null;
+  }
+
+  _getNumberFormatter(lang) {
```

Comment:
nit: can you move this to the bottom of the file, closer to `_getEntity`, please?